modules/client-common/src/utils.ts

```diff
@@ -199,6 +211,169 @@ export async function* prepareGenericInstallation(
     helpers: preparedSetupData.helpers,
   };
 }
+export async function prepareGenericUpdateEstimation(
+  web3: IClientWeb3Core,
+  graphql: IClientGraphQLCore,
+  params: PrepareUpdateParams & { pluginSetupProcessorAddress: string },
+): Promise<GasFeeEstimation> {
+  const signer = web3.getConnectedSigner();
+  type T = {
+    iplugin: { installations: SubgraphPluginInstallation[] };
+  };
+  const { iplugin } = await graphql.request<T>({
+    query: QueryIPlugin,
+    params: {
+      address: params.pluginAddress.toLowerCase(),
+      where: { dao: params.daoAddressOrEns.toLowerCase() },
+    },
+    name: "plugin",
+  });
+
+  // filter specified installation
+  const { pluginInstallationIndex = 0 } = params;
+  const selectedInstallation = iplugin.installations[pluginInstallationIndex];
+  if (!selectedInstallation) {
+    throw new InstallationNotFoundError();
+  }
+  // check if version is valid
+  if (
+    params.newVersion.release !==
+      selectedInstallation.appliedVersion.release.release ||
+    params.newVersion.build <= selectedInstallation.appliedVersion.build
+  ) {
+    throw new InvalidVersionError();
+  }
+  // encode update params
+  const { updateParams = [], updateAbi = [] } = params;
+  const data = defaultAbiCoder.encode(
+    getNamedTypesFromMetadata(updateAbi),
+    updateParams,
+  );
+  // connect to psp contract
+  const pspContract = PluginSetupProcessor__factory.connect(
+    params.pluginSetupProcessorAddress,
+    signer,
+  );
+  const prepareUpdateParams = {
+    currentVersionTag: {
+      build: selectedInstallation.appliedVersion.build,
+      release: selectedInstallation.appliedVersion.release.release,
+    },
+    newVersionTag: params.newVersion,
+    pluginSetupRepo: params.pluginRepo,
+    setupPayload: {
+      plugin: params.pluginAddress,
+      currentHelpers: selectedInstallation.appliedPreparation.helpers,
+      data,
+    },
+  };
+
+  const gasEstimation = await pspContract.estimateGas.prepareUpdate(
+    params.daoAddressOrEns,
+    prepareUpdateParams,
+  );
+  return web3.getApproximateGasFee(gasEstimation.toBigInt());
+}
+
+export async function* prepareGenericUpdate(
```

Comment:
move code from here and `prepareGenericUpdateEstimation()` that is duplicated into another utils function to reduce code duplication